### PR TITLE
OCM-9776 | test: Automate OCP-55883 Upgrade STS rosa cluster across Y stream via ROSA cli

### PR DESCRIPTION
**Pull request by jameszwang**

### CODE REVIEW

File: tests/e2e/test_rosacli_upgrade.go [hunk 1]
```diff
@@ -449,6 +504,25 @@ var _ = Describe("Describe/List rosa upgrade",
 			})
 	})
 
+func FindUpgradeVersion(v rosacli.VersionService, channelGroup string, hostedCP bool, clusterVersion string) (string,
```
Comment (yasun1) [inline tests/e2e/test_rosacli_upgrade.go:507]: IMO, we should do the Y-stream upgrading in the process y->y+1->y+2. So we should find the latest y+1 version.

File: tests/e2e/test_rosacli_upgrade.go [hunk 2]
```diff
@@ -314,6 +309,66 @@ var _ = Describe("Cluster Upgrade testing",
...
+					"-y",
+				)
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring("Upgrade successfully scheduled for cluster"))
+			}
+
+			By("Check upgrade state")
+			err = WaitForUpgradeToState(upgradeService, clusterID, con.Scheduled, 4)
+			Expect(err).To(BeNil())
+			err = WaitForUpgradeToState(upgradeService, clusterID, con.Started, 70)
```
Comment (yasun1) [inline tests/e2e/test_rosacli_upgrade.go:369]: Do we not wait for the cluster upgrading to be finished?

File: tests/e2e/test_rosacli_upgrade.go (hunk 1 above)
Comment (jameszwang) [reply tests/e2e/test_rosacli_upgrade.go]: Update PR according to our discussion, sorted upgrade path versions.

File: tests/e2e/test_rosacli_upgrade.go (hunk 2 above)
Comment (jameszwang) [reply tests/e2e/test_rosacli_upgrade.go:372]: According to our discussion, we verify the upgrade state currently and will enhance this part in the future.

File: tests/e2e/test_rosacli_upgrade.go [hunk 3]
```diff
@@ -449,6 +507,81 @@ var _ = Describe("Describe/List rosa upgrade",
...
+
+func FindUpperYStreamVersion(v rosacli.VersionService, channelGroup string, hostedCP bool,
+	clusterVersion string) (string, string, error) {
+	clusterVersionList, err := v.ListAndReflectVersions(channelGroup, hostedCP)
+	if err != nil {
+		return "", "", err
+	}
+	versions, err := clusterVersionList.FindYStreamUpgradeVersions(clusterVersion)
+	// Sorted version from high to low
+	sortedVersions := SortVersions(versions)
```
Comment (yasun1) [inline tests/e2e/test_rosacli_upgrade.go:571]: This line should be put after 574.

File: tests/e2e/test_rosacli_upgrade.go [hunk 4]
```diff
@@ -449,6 +507,81 @@ var _ = Describe("Describe/List rosa upgrade",
...
+	for i, v := range versions {
+		parsedVersions[i] = ParseVersion(v)
+	}
+	sort.Sort(sort.Reverse(VersionSlice(parsedVersions)))
+	sortedVersions := make([]string, len(versions))
+	for i, v := range parsedVersions {
+		sortedVersions[i] = v.Original
+	}
+	return sortedVersions
+}
```
Comment (yasun1) [inline tests/e2e/test_rosacli_upgrade.go:561]: We have had `sort` functions https://github.com/openshift/rosa/blob/master/tests/utils/exec/rosacli/version_service.go#L221 for this. You just need to use it directly in `FindUpperYStreamVersion`.

**Pull request merged**